pkg/federate/base_federator.go

```diff
@@ -65,9 +72,26 @@ func (f *baseFederator) Delete(obj runtime.Object) error {
 		return err
 	}
 
+	ctx := context.TODO()
+
+	if toDelete.GetName() == "" && toDelete.GetGenerateName() != "" {
```

Comment:
The Kubernetes APIs don’t support `GenerateName` on delete operations, do we really need this?

Reply:
This API isn't trying to mimic/mirror the K8s delete API. The scenario here is the caller wants to delete a resource where a well-known name isn't known, ie it was generated via `GenerateName`.  The motivating use case is for transforming K8s `EndpontSlices`  to LH `EndpontSlices` via a resource syncer. Instead of constructing a well-known name by concatenating various fields and risking exceeding the max length,  we use the `GenerateName` functionality. However, on update and delete, this means we need to find the target resource via `List` by its unique labels derived from the source resource. We could do this in the caller but I thought to generalize it here and in `CreateOrUpdate` and codify the intention implicitly in the passed resource. We could codify it in the function signatures instead to make it  explicit. The other option is to have the caller obtain the target resource and not change anything in admiral.

Reply:
Ah right, that makes sense. My main concern then with the generalisation is that it should apply generally — it might be sufficient to return an error if the `List` calls return more than one result...

Reply:
@skitt and I discussed offline - we don't think handling `GenerateName` semantics fits here so we'll let the client handle the lookup and set the name.

Reply:
This code was removed.